pokemon-service/src/test/java/cz/muni/fi/pa165/pokemon/league/participation/manager/service/GymServiceTest.java

```diff
@@ -0,0 +1,376 @@
+                .location("Praha")
+                .build();
+
+        gymBrno = new GymBuilder()
+                .id(11L)
+                .gymLeader(gymLeaderBrno)
+                .type(PokemonType.FIGHTING)
+                .location("Brno")
+                .build();
+
+        gymOstrava = new GymBuilder()
+                .id(11L)
+                .gymLeader(gymLeaderOstrava)
+                .type(PokemonType.GROUND)
+                .location("Ostrava")
+                .build();
+
+        errorneousGym = new GymBuilder()
+                .id(Long.MAX_VALUE)
+                .gymLeader(gymLeaderOstrava)
+                .type(PokemonType.BUG)
+                .location("Nowhere")
+                .build();
+
+        when(gymDAO.findGymById(gymPraha.getId())).thenReturn(gymPraha);
+        when(gymDAO.findGymById(gymBrno.getId())).thenReturn(gymBrno);
+        when(gymDAO.getAllGyms()).thenReturn(Stream.of(gymPraha, gymBrno)
+                .collect(Collectors.toList()));
+
+    }
+
+    @Test
+    public void testFindById() {
+        assertThat(gymService.findGymById(gymPraha.getId()))
+                .isEqualToComparingFieldByField(gymPraha);
+    }
+
+    @Test
+    public void testFindByIdWithException() {
+        doThrow(dae).when(gymDAO).findGymById(errorneousGym.getId());
+
+        assertThatExceptionOfType(DataAccessException.class)
+                .as("Testing DAO exception")
+                .isThrownBy(() -> gymService.findGymById(errorneousGym.getId()));
+    }
+
+    @Test
+    public void testFindByNonExistingId() {
+        assertThat(gymService.findGymById(Long.MIN_VALUE)).isNull();
+    }
+
+    @Test
+    public void testUpdateGymLocation() {
+        String newLocation = "New Location";
+        gymService.updateGymLocation(gymPraha, newLocation);
+
+        assertThat(gymService.findGymById(gymPraha.getId()).getLocation())
+                .isEqualTo(newLocation);
+    }
+
+    @Test
+    public void testUpdateGymLocationWithException() {
+        doThrow(dae).when(gymDAO).updateGym(errorneousGym);
+
+        String newLocation = "New Location";
+
+        assertThatExceptionOfType(DataAccessException.class)
+                .as("Testing DAO exception")
+                .isThrownBy(() -> gymService.updateGymLocation(errorneousGym, newLocation));
+    }
+
+    @Test
+    public void testChangeGymTypeByOwnTrainer() {
+        PokemonType newType = PokemonType.STEEL;
+        boolean result = gymService.changeGymType(gymPraha, gymPraha.getGymLeader(), newType);
+        assertThat(result).isTrue();
+        assertThat(gymService.findGymById(gymPraha.getId()).getType())
+                .isEqualTo(newType);
+    }
+
+    @Test
+    public void testChangeGymTypeByOtherTrainer() {
+        PokemonType originalType = gymPraha.getType();
+        PokemonType newType = PokemonType.STEEL;
+        boolean result = gymService.changeGymType(gymPraha, otherTrainer, newType);
+        assertThat(result).isFalse();
```

Comment:
Why not condense the service call into the assertThat? This will probably change into a check for InsufficientRightsException anyway.